packages/nx/src/tasks-runner/tasks-schedule.ts

```diff
@@ -112,12 +121,26 @@ export class TasksSchedule {
         const project1 = this.taskGraph.tasks[taskId1].target.project;
         const project2 = this.taskGraph.tasks[taskId2].target.project;
 
-        return (
-          findAllProjectNodeDependencies(project2, this.reverseProjectGraph)
-            .length -
-          findAllProjectNodeDependencies(project1, this.reverseProjectGraph)
-            .length
-        );
+        const project1NodeDependencies = findAllProjectNodeDependencies(
+          project1,
+          this.reverseProjectGraph
+        ).length;
+        const project2NodeDependencies = findAllProjectNodeDependencies(
+          project2,
+          this.reverseProjectGraph
+        ).length;
+
+        const dependenciesDiff =
+          project2NodeDependencies - project1NodeDependencies;
+
+        if (dependenciesDiff !== 0) {
+          return dependenciesDiff;
+        }
+
+        const task1Timing = this.estimatedTaskTimings[taskId1] ?? 0;
+        const task2Timing = this.estimatedTaskTimings[taskId2] ?? 0;
+
+        return task2Timing - task1Timing;
```

Comment:
Change this so that if a task does not have an estimated timing, we assume it is a new task, which should yield more interesting results... thus, we should run those first.